internal/services/network/network_manager_routing_configuration_resource.go

```diff
@@ -0,0 +1,198 @@
+			}
+
+			networkManagerId, err := networkmanagerroutingconfigurations.ParseNetworkManagerID(config.NetworkManagerId)
+			if err != nil {
+				return err
+			}
+
+			id := networkmanagerroutingconfigurations.NewRoutingConfigurationID(subscriptionId, networkManagerId.ResourceGroupName, networkManagerId.NetworkManagerName, config.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := networkmanagerroutingconfigurations.NetworkManagerRoutingConfiguration{
+				Name: pointer.To(config.Name),
+				Properties: &networkmanagerroutingconfigurations.NetworkManagerRoutingConfigurationPropertiesFormat{
+					Description: pointer.To(config.Description),
+				},
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, payload); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r ManagerRoutingConfigurationResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Network.NetworkManagerRoutingConfigurations
+
+			id, err := networkmanagerroutingconfigurations.ParseRoutingConfigurationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			networkManagerId := networkmanagerroutingconfigurations.NewNetworkManagerID(id.SubscriptionId, id.ResourceGroupName, id.NetworkManagerName).ID()
+			schema := ManagerRoutingConfigurationResourceModel{
+				Name:             id.RoutingConfigurationName,
+				NetworkManagerId: networkManagerId,
+			}
+
+			if model := resp.Model; model != nil && model.Properties != nil {
+				schema.Description = pointer.From(model.Properties.Description)
+			}
+
+			return metadata.Encode(&schema)
+		},
+	}
+}
+
+func (r ManagerRoutingConfigurationResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Network.NetworkManagerRoutingConfigurations
+
+			id, err := networkmanagerroutingconfigurations.ParseRoutingConfigurationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model ManagerRoutingConfigurationResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			parameters := networkmanagerroutingconfigurations.NetworkManagerRoutingConfiguration{
```

Comment:
since `client.CreateOrUpdate` is used, `parameters` should be based on the response of `client.Get`?

Reply:
fixed